scripts/helpers/restyle-diff.sh

```diff
@@ -41,6 +41,17 @@ restyle-paths() {
 cd "$CHIP_ROOT"
 declare -a paths=()
 
-readarray -t paths < <(git diff --ignore-submodules --name-only "${1:-master}")
+if [[ "$OSTYPE" == "darwin"* ]]; then
+    (git diff --ignore-submodules --name-only "${1:-master}") >./paths.txt
+
+    paths=()
+    while IFS= read -r line; do
+        paths+=("$line")
+    done <./paths.txt
+
+    rm ./paths.txt
+else
+    readarray -t paths < <(git diff --ignore-submodules --name-only "${1:-master}")
```

Comment:
Does this work?

`declare -a paths=($(git diff --ignore-submodules --name-only "${1:-master}"))`

Reply:
also wonder if we could make the same method work on both platforms (worst case move this to python)

Reply:
@mspang I will test it, i don't know if that works
@andy31415 The method I added for darwin would probably work on linux too. The readarray method is probably more efficent and is cleaner.

Reply:
@mspang It does work with added quotes
 declare -a paths="($(git diff --ignore-submodules --name-only "${1:-master}"))"

I just need to make sur it works on linux also